### PR TITLE
simpleRestJson: add protocol tests for default body behavior

### PR DESCRIPTION
**Pull request by ghostbuster91**
Adding missing tests for a case that was reported in https://github.com/disneystreaming/smithy4s/issues/1502

### CODE REVIEW

File: modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+namespace alloy.test
+
+use alloy#simpleRestJson
+use smithy.test#httpRequestTests
+use smithy.test#httpResponseTests
+
+apply HttpPayloadWithDefault @httpRequestTests([{
+    id: "SimpleRestJsonSomeHttpPayloadWithDefault"
+    documentation: "Pass simple string value as is if payload provided"
```
Comment (kubukoz) [inline modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy:11]: **suggestion:** 
```suggestion
    documentation: "Pass JSON string value as is if payload provided"
```

File: modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        "Content-Type": "application/json"
+    }
+    params: {
+        body: "default value"
+    }
+}])
+
+apply HttpPayloadWithDefault @httpResponseTests([{
+    id: "SimpleRestJsonSomeHttpPayloadWithDefault"
+    documentation: "Pass simple string value as is if payload provided"
```
Comment (kubukoz) [inline modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy:40]: **suggestion:** 
```suggestion
    documentation: "Pass JSON string value as is if payload provided"
```

**Review by kubukoz — APPROVED** [review]

File: modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    bodyMediaType: "application/json"
+}, {
+    id: "SimpleRestJsonNoneHttpPayloadWithDefault"
+    documentation: "Use default value when there is no payload"
+    protocol: simpleRestJson
+    method: "PUT"
+    uri: "/httpPayloadWithDefault"
+    headers: {
+        "Content-Type": "application/json"
+    }
```
Comment (kubukoz) [inline modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy:32]: **issue:** hmm I think you'll have to skip the content-type when there's no body: it's not valid JSON to send nothing

File: modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        body: "custom value"
+    }
+}, {
+    id: "SimpleRestJsonNoneHttpPayloadWithDefault"
+    documentation: "Use default value when there is no payload"
+    protocol: simpleRestJson
+    code: 200
+    headers: {
+        "Content-Type": "application/json"
+    }
```
Comment (kubukoz) [inline modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy:58]: **issue:** same as above

**Review by kubukoz — CHANGES_REQUESTED** [review]

File: modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy (hunk 3 above)
Comment (ghostbuster91) [reply modules/protocol-tests/resources/META-INF/smithy/HttpPayloadWithDefault.smithy]: done

**Review by kubukoz — APPROVED** [review]

**Pull request merged**